packages/dma-contracts/contracts/actions/morpho-blue/Payback.sol

```diff
@@ -32,7 +38,24 @@ contract MorphoBluePayback is Executable, UseStore {
     IMorpho morphoBlue = IMorpho(registry.getRegisteredService(MORPHO_BLUE));
 
     address onBehalf = paybackData.onBehalf == address(0) ? address(this) : paybackData.onBehalf;
-    morphoBlue.repay(paybackData.marketParams, paybackData.amount, 0, onBehalf, bytes(""));
+
+    if (paybackData.paybackAll) {
+      Id id = paybackData.marketParams.id();
+
+      // Need to call accrueInterest to get the latest snapshot of the shares/asset ratio
+      morphoBlue.accrueInterest(paybackData.marketParams);
+
+      uint256 totalBorrowAssets = morphoBlue.totalBorrowAssets(id);
+      uint256 totalBorrowShares = morphoBlue.totalBorrowShares(id);
+      uint256 shares = morphoBlue.borrowShares(id, onBehalf);
+      uint256 assetsMax = shares.toAssetsUp(totalBorrowAssets, totalBorrowShares);
+
+      require(paybackData.amount >= assetsMax, "MorphoBluePayback: payback amount too low");
+
+      morphoBlue.repay(paybackData.marketParams, 0, shares, onBehalf, bytes(""));
+    } else {
+      morphoBlue.repay(paybackData.marketParams, paybackData.amount, 0, onBehalf, bytes(""));
+    }
 
     store().write(bytes32(paybackData.amount));
```

Comment:
Can you please write to the storage actual payback amount not the amount passed to the action.